contracts/programs/gummyshop/src/lib.rs

```diff
@@ -0,0 +1,430 @@
+    #[account(
+        seeds = [MARKETPLACE_PROPERTIES_PREFIX.as_ref()],
+        bump = marketplace_props.bump,
+    )]
+    #[account(mut)]
+    pub marketplace_props: Account<'info, MarketplaceProperties>,
+    pub system_program: Program<'info, System>
+}
+
+#[derive(Accounts)]
+pub struct WithdrawFees<'info> {
+    /// CHECK: any pubkey the marketplace wants to withdraw fees to
+    #[account(mut)]
+    pub fee_payout_recipient: AccountInfo<'info>,
+
+    /// CHECK: the authority over the marketplace, must correspond to the authority in marketplace_props, validated in instruction.
+    pub authority: Signer<'info>,
+
+    /// CHECK: the PDA for this marketplace to withdraw fees from.
+    #[account(
+        seeds = [MARKETPLACE_PROPERTIES_PREFIX.as_ref()],
+        bump = marketplace_props.bump,
+    )]
+    #[account(mut)]
+    pub marketplace_props: Account<'info, MarketplaceProperties>,
+    pub system_program: Program<'info, System>,
+    pub sysvar_rent: Sysvar<'info, Rent>
+}
+
+// A helper function to CPI to Bubblegum delegate. Used for listing creation, modification and removal.
+#[inline(always)]
+fn modify_compressed_nft_delegate<'info>(
+    owner: &Signer<'info>,
+    former_delegate: &AccountInfo<'info>,
+    new_delegate: &AccountInfo<'info>,
+    bubblegum_authority: &AccountInfo<'info>,
+    gummyroll: &Program<'info, Gummyroll>,
+    merkle_slab: &AccountInfo<'info>,
+    bubblegum: &Program<'info, Bubblegum>,
+    data_hash: [u8; 32],
+    creator_hash: [u8; 32],
+    nonce: u64,
+    index: u32,
+    root: [u8; 32]
+) -> Result<()> {
+    let cpi_ctx = CpiContext::new(
+        bubblegum.to_account_info(),
+        bubblegum::cpi::accounts::Delegate {
+            authority: bubblegum_authority.to_account_info(),
+            owner: owner.to_account_info(),
+            previous_delegate: former_delegate.to_account_info(),
+            new_delegate: new_delegate.to_account_info(),
+            gummyroll_program: gummyroll.to_account_info(),
+            merkle_slab: merkle_slab.to_account_info(),
+        }
+    );
+    bubblegum::cpi::delegate(cpi_ctx, root, data_hash, creator_hash, nonce, index)?;
+    Ok(())
+}
+
+#[program]
+pub mod gummyshop {
+    use super::*;
+
+    /// Initialize the singleton PDA that will store the marketplace's admin info, mainly related to royalties.
+    pub fn initialize_marketplace(
+        ctx: Context<InitMarketplaceProperties>,
+        royalty_share: u16,
+        authority: Pubkey
+    ) -> Result<()> {
+        let marketplace_props_data = &mut ctx.accounts.marketplace_props;
+        assert!(royalty_share <= 10000);
+        marketplace_props_data.share = royalty_share;
+        marketplace_props_data.authority = authority;
+        marketplace_props_data.bump = *ctx.bumps.get("marketplace_props").unwrap();
+        Ok(())
+    }
+
+    /// Enables the authority of the marketplace to update admin properties 
+    pub fn update_marketplace_properties<'info>(
```

Comment:
There aren't currently tests for this instruction because I wasn't sure if it would be desired, but can add if it sticks.

imo benefits:
- authority can be changed (good for key leaks, etc.)
- share can change (good for marketplace flexibility)

imo cons:
- people may list at fee x, and then have fee y taken when the marketplace changes their props. Could be unexpected to the lister. 

For an example implementation it might be worth having, just to show marketplaces the most flexibility, curious about your thoughts @jarry-xiao